### PR TITLE
New module_view, module_list_view & xAPI events

### PR DESCRIPTION
**Pull request by nadavkav**

### CODE REVIEW

File: ajax.php [hunk 1]
```diff
@@ -251,6 +251,33 @@
...
+
+        $event = \mod_hvp\event\hvp_xapi::create(array(
+            'objectid' => $hvpid,
+            'context' => $context,
+            'other' => array(
+                'actor' => $xapistatement['data']['statement']['actor'],
+                'verb' => $xapistatement['data']['statement']['verb'],
+                'object' => $xapistatement['data']['statement']['object'],
+                'context' => $xapistatement['data']['statement']['context'],
+                'result' => $xapistatement['data']['statement']['result']),
```
Comment (thomasmars) [inline ajax.php]: Result is not always part of an xAPI event, for instance 'interacted' verb does not provide a result. Getting undefined php error.

**Comment by thomasmars** [comment]
Still getting excessive amount of views if I go to the view of an H5P and perform some xAPI events. Seems I get 1 view for each xAPI event fired.

**Comment by thomasmars** [comment]
Also, I am wondering about the package.json, is there a reason for adding it to the module ? What purpose does it serve ?

**Comment by thomasmars** [comment]
And debugging lines should be removed before merging this in.

File: ajax.php (hunk 1 above)
Comment (nadavkav) [inline ajax.php]: I see.
What about just adding:
 'statement' => $xapistatement['data']['statement']

Comment (thomasmars) [inline ajax.php]: Yeah, sounds reasonable to me

**Comment by nadavkav** [comment]
The **package.json** is required to "[compile](https://docs.moodle.org/dev/Grunt#Using_Grunt_in_additional_plugins)" js amd files (I added) using Grunt.

**Comment by nadavkav** [comment]
I can easily hide it. but...
The "debugging" is not a real debugging and more like a feedback which can help a developer (or an advanced user) that uses the Browser's developer tools to view and verify that the AJAX request to log the xAPI event was successful. A regular user do not see it.
So, should I remove/hide it?

**Comment by nadavkav** [comment]
Regarding "Still getting excessive amount of views if I go to the view of an H5P and perform some xAPI events. Seems I get 1 view for each xAPI event fired."...
Can you please remove the H5P (hvp) module and reinstall it, 
As I can not reproduce it, on two Moodle systems I already use this code. (btw, they are both bind to an external LRS using the xAPI logstore plugin, which too, does not show redundant "view" events)

**Comment by thomasmars** [comment]
Regarding the package.json file. Correct me if I'm wrong, but the grunt configuration file is not added to the module, so including the package.json will not help other developers build the AMD files with Grunt. Also the rest of the scripts does not use any build tools, so I don't really see the need to introduce it for just one file, I guess it is in order to include the dependency to core/log ? Is there an alternative way of doing this that will allow us to skip the build step ? If not Grunt configuration should be added and dev dependencies should be included in package.json, so any developer working on this has everything he needs out of the box.

**Comment by thomasmars** [comment]
In my mind best practice would be to remove debugging lines before a product is production ready, however I don't know about Moodle, perhaps it is okay to leave it there. The console log at xAPI post success seems superfluous to me at least.

**Comment by thomasmars** [comment]
I tried reinstalling Moodle and the H5P plugin, and I am still getting too many 17 views when navigating once to the H5P view.php from the activity report and then performing several xAPI events, such as attempted and completed. I was able to reproduce this in "Fill in the Blanks", can you try this one more time and see if you can reproduce ? :)

**Comment by nadavkav** [comment]
Grunt is highly advised to be used by Moodle developers. but I am not sure it is a must. as you can run jshit uglify, rebuild all AMD modules (I am using it for some time, so I was not aware it was an issue). Unless a plugin developer needs a special gruntfile.js, the one in the Moodle's root is used. and that is way I did not add and special gruntfile.js, just the package.json.
Moodle is moving from YUI only to jQuery and other JS frameworks using AMD, and so AMD is highly advised (maybe, required) to be used in any new code & plugins.  I guess I could use a simple JS and not use jQuery or core/log but it seems I must use AMD. so why not.

I have conditioned debug output on Moodle's debug settings.

**Comment by nadavkav** [comment]
THis PR is deprecated in favor of PR #89

**Comment by nadavkav** [comment]
@falcon-git / @icc , Please see if you can use the new patch at PR #89 and see if you also get redundant "view" events when "xAPI" events are fired.

**Pull request closed without merging**